### DISCUSSION TITLE
Help with Monero config.json - System.NullReferenceException: Object reference not set to an instance of an object.

### DISCUSSION THREAD

**Discussion by bdk773** [coinfoundry/miningcore#905] [Q&A] [answered]
Followed the guide to the dot, getting the following error:

> 
> [2021-07-20 14:49:21.7517] [I] [Core] .NET 5.0.8 on Microsoft Windows 10.0.17763 [X64]
> [2021-07-20 14:49:22.9823] [I] [Core] Prometheus Metrics 127.0.0.1:4000/metrics
> [2021-07-20 14:49:22.9823] [I] [Core] WebSocket notifications streaming 127.0.0.1:4000/notifications
> [2021-07-20 14:49:28.1302] [I] [XmlKeyManager] User profile is available. Using 'C:\Users\Administrator\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
> [2021-07-20 14:49:28.2986] [I] [ShareRecorder] Online
> [2021-07-20 14:49:28.5238] [I] [PayoutManager] Online
> [2021-07-20 14:49:28.5238] [I] [StatsRecorder] Online
> [2021-07-20 14:49:28.8448] [I] [Core] 52 coins loaded from C:\Program Files\miningcore\coins.json
> [2021-07-20 14:49:29.1889] [I] [xmr1] Starting Pool ...
> [2021-07-20 14:49:29.2518] [E] [xmr1] System.NullReferenceException: Object reference not set to an instance of an object.
> 
> xmr1 pointing of course to my pools portion of the config.json (Sample take from the miningcore repo minus minor adjustments such as wallet address etc..)
> "pools": [{
>       "id": "xmr1",

Per caveat mentioned in the main repo bypass (--disable-rpc-login) was implemented as well as username:pw empty for daemon:

> "daemons": [
>         {
>           "host": "127.0.0.1",
>           "port": 18081,
>            "user": null,
>           "password": null
>        },
>         {
>           "host": "127.0.0.1",
>           "port": 18082,
>           "user": null,
>           "password": "password123",
>           "category": "wallet"
>         }
>       ],

Testing performed on windows 2019 (fully patched), 13.3.2 sql, dotnet 5.0.8. Please advise if anybody encountered this error before?! Thank you in advance.

PS> Entire log below:
https://pastecode.io/s/z49d6s95

**Comment by ahmedbodi** [comment]
Add ```'randomXRealm': 'xmr1'``` to your config for monero

**Comment by calvintam236** [comment, marked as answer]
See #902.
